pkg/rulefmt/rulefmt.go

```diff
@@ -137,6 +142,58 @@ func (r *Rule) Validate() (errs []error) {
 		}
 	}
 
+	errs = append(errs, testTemplateParsing(r)...)
+	return errs
+}
+
+// testTemplateParsing checks if the templates used in labels and annotations
+// of the alerting rules are parsed correctly.
+// NOTE: It wont detect if a wrong key is used with the variables, like `$labels.wrongKey`.
+//       This will check for wrong variables used, like `$wrongLabelsVariable`.
+//       Also wrong usage of template like {{$labels.quantile * 100}}, etc.
+func testTemplateParsing(rl *Rule) (errs []error) {
+	if rl.Alert == "" {
+		// Not an alerting rule.
+		return errs
+	}
+
+	// Trying to parse templates.
+	tmplData := struct {
```

Comment:
We seem to be duplicating a fair bit of code here, so there might be drift. Could this be refactored to avoid that?

Reply:
@brian-brazil I added a function for that in `template.go` and using that here and in `alerting.go`. Does it look right?